_source/_posts/2023-01-24-jakartaee-auth0.md

```diff
@@ -0,0 +1,545 @@
+
+To run an SE application, all you need is the JRE (Java runtime environment) for a compatible version of Java. Enterprise Edition, however, requires a more complete runtime environment and has a lot more possible modules and configuration options. To see a list of Jakarta EE compatible products, you can look at [the Jakarta website](https://jakarta.ee/compatibility/). A few examples are Open Liberty, Payara, Wildfly, Glassfish, and TomEE.
+
+As of the time I wrote this tutorial, Jakarta EE 10 was a very new release, and only three frameworks supported version 10: Eclipse GlashFish, Payara Server Community, and WildFly. 
+
+**Wildfly**
+
+I chose to use [WildFly](https://www.wildfly.org/) as my Jakarta EE runtime. Hantsy Bai created a great example project that was a big help. Check out [the GitHub project repository page](https://github.com/hantsy/jakartaee10-sandbox). Thanks, Hantsy Bai! Super helpful.
+
+WildFly is an open-source community project sponsored by Red Hat. It bills itself as a "flexible, lightweight, managed application runtime" that is "based on Jakarta EE and provides rich enterprise capabilities in easy to consume frameworks that eliminate boilerplate and reduce technical burden." It is a modular, standards-based runtime for Jakarta EE applications.
+
+**Jakarta EE 10 Security 3.0**
+
+The exciting thing about Jakarta 10 (from a security perspective) is that it includes a new OIDC implementation in the Security 3.0 specification. Open ID Connect is an authentication protocol. This protocol is implemented by many third-party vendors, such as Auth0 and Okta, making it relatively easy to add secure login to an application. Jakarta EE 10 Security 3.0 provides an annotation-based configuration to add OIDC authentication to servlets.
+
+You can check out [the docs for Jakarta EE 10 Security 3.0 OIDC here](https://jakarta.ee/specifications/security/3.0/jakarta-security-spec-3.0.html#openid-connect-annotation).
+
+**Requirements**
+
+Before you start, please make sure you have the following prerequisites installed (or install them now).
+
+- [Java 17](https://adoptium.net/): or use [SDKMAN!](https://sdkman.io/) to manage and install multiple versions (the Jakarta EE spec says 11 and up is supported, but I wrote this tutorial assuming version 17)
+- [Auth0 CLI](https://github.com/auth0/auth0-cli#installation): the Auth0 command-line interface
+- [HTTPie](https://httpie.org/doc#installation): a simple tool for making HTTP requests from a Bash shell
+
+**You will need a free Auth0 developer account** if you don't already have one. Go ahead and sign up for an Auth0 account using [their sign-up page](https://auth0.com/signup).
+
+Clone the tutorial from [the GitHub repository](need.a.link).
+
+## Take a look at the build configuration and project dependencies
```

Comment:
If you look at the preview on Netlify, you'll see this is quite long. You can add a TOC at the beginning to better visualize all the headlines. 

```
{% include toc.md %}
```

Reply:
OK. I removed some of it. Sometimes I get into the weeds a little.